examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java

```diff
@@ -106,6 +107,43 @@ public static void main(String[] args) {
     spark.stop();
   }
 
+  private static void runGenericFileSourceOptionsExample(SparkSession spark) {
+    // $example on:ignore_corrupt_files$
+    // enable ignore corrupt files
+    spark.sql("set spark.sql.files.ignoreCorruptFiles=true");
+    // dir1/file3.json is corrupt from parquet's view
+    Dataset<Row> testCorruptDF = spark.read().parquet(
+            "examples/src/main/resources/dir1/",
+            "examples/src/main/resources/dir1/dir2/");
+    testCorruptDF.show();
+    // +-------------+
+    // |         file|
+    // +-------------+
+    // |file1.parquet|
+    // |file2.parquet|
+    // +-------------+
+    // $example off:ignore_corrupt_files$
+    spark.sql("set spark.sql.files.ignoreMissingFiles=false");
+    // $example on:load_with_path_glob_filter$
+    Dataset<Row> partitionedUsersDF = spark.read().format("orc")
+            .option("pathGlobFilter", "*.orc")
+            .load("examples/src/main/resources/partitioned_users.orc");
```

Comment:
shall we show the result like the previous example?

Reply:
this is from an existing example. I think we should rewrite it to use the same file set. How about using `*.parquet`?

Reply:
and see if we can remove `partitioned_users.orc`

Reply:
After checking the partitioned_users.orc, it's a demo for hive partitioned table, add the description for `partitioned_users.orc` in the doc. Done in 1de38be.